Lesson4/1.js

```diff
@@ -0,0 +1,10 @@
+/*
+Модифицировать прототип Array, добавив в него метод shuffle, 
+чтобы получить возможность выполнять случайную сортировку для любого массива
+*/
+
+Array.prototype.shuffle = function(){return this.sort(function() { return Math.random() - Math.random(); })};
```

Comment:
Можно немного упростить вычисления и предотвратить перемешивание исходного массива:
```suggestion
Array.prototype.shuffle = function(){return [...this].sort(() => Math.random() - 0.5)};
```

```js
const array = [1, 2, 3];

console.debug(array.shuffle()); // [2, 1, 3]

console.debug(array); // [1, 2, 3]
```